src/posix/routing_manager/router_advertisement.cpp

```diff
@@ -0,0 +1,481 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes implementations for IPv6 Router Advertisement.
```

Comment:
How about having a platform-independent implementation for generating and processing ICMPv6 Router Advertisements in core? This helps reduce the platform-specific code to just sending/receiving fully-formed ICMPv6 RA messages. This can help porting to non-POSIX platforms. It also allows full access to OpenThread utilities, such as timer. Thoughts?

Reply:
@jwhui Thanks for the suggestion, sounds attractive! Then we need to define platform API for the infrastructure interface to:
- send, receive ICMPv6 messages on this infra interface
- get notified about the state of the infra interface so that we may start/stop routing manager when the infra interface becomes up/down.

example APIs in `include/openthread/platform` could be like:
```shell
otError otInfraLinkRecvIcmp6(const char *aIfName, const uint8_t *aMessage, uint16_t aLength);
otError otInfraLinkSendIcmp6(const char *aIfName, const uint8_t *aMessage, uint16_t aLength);
otError otInfraLinkSetUp(const char *aIfName, bool aIsUp);
```

Will firstly get the CI tests and then refactor this initial commit.